rest/rest-server/src/main/java/org/apache/brooklyn/rest/RestApiSetup.java

```diff
@@ -54,6 +63,21 @@ public static void installRest(ServletContextHandler context, Object... provider
         context.addServlet(servletHolder, "/v1/*");
     }
 
+    private static void injectManagementContext(ManagementContext managementContext, Object o) {
+        if (managementContext != null && o instanceof ManagementContextInjectable) {
+            ((ManagementContextInjectable) o).setManagementContext(managementContext);
+        }
+    }
+
+    private static ManagementContext extractManagementContext(Object... providers) {
+        for (Object o : providers) {
+            if (o instanceof ManagementContextProvider) {
+                return ((ManagementContextProvider) o).getContext(ManagementContext.class);
+            }
+        }
+        return null;
+    }
+
```

Comment:
This will break when combined with #41 because `ManagementContextProvider` no longer holds a reference to the `ManagementContext` in the classic launcher case. Instead it fetches it from `ServletContext`.
This change is needed because of the additional `managementContextInternal` change. Instead I suggest letting `ManagementContextProvider` handle the injection in resources (i.e. adding @Context .... to the property). This way we will have a unified code path in both classical launcher and OSGi.